nilearn/plotting/displays/_slicers.py

```diff
@@ -416,18 +433,20 @@ def _map_show(
         # case where this image is binary, such as when this function
         # is called from `add_contours`, continuous interpolation
         # does not make sense and we turn to nearest interpolation instead.
+        transparency, transparency_affine = self._sanitize_transparency(
+            img, transparency, transparency_range, resampling_interpolation
+        )
+
         if is_binary_niimg(img):
             img = reorder_img(img, resample="nearest", copy_header=True)
         else:
             img = reorder_img(
                 img, resample=resampling_interpolation, copy_header=True
             )
+
+        #  TODO resample transparency to input image ???
+
         affine = img.affine
-        if threshold is not None:
-            threshold = float(threshold)
-            data = safe_get_data(img, ensure_finite=True)
-            data = self._threshold(data, threshold, None, None)
-            img = new_img_like(img, data, affine)
```

Comment:
I do not believe this is needed as each slice is threholded before calling draw_2d